tools/toolchains/__init__.py

```diff
@@ -284,7 +284,8 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         self.silent = silent
         
         # Print output buffer
-        self.output = ""
+        self.output = str()
+        self.map_outputs = list()   # Place to store memmap scan results in JSON like data structures
```

Comment:
Again, this could be collected across all compiles similar to add_result_to_report() which is already used for tests.